WalletWasabi/Wallets/FilterProcessor/BlockFilterIterator.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Backend.Models;
+using WalletWasabi.Stores;
+
+namespace WalletWasabi.Wallets.FilterProcessor;
+
+/// <summary>
+/// Iterator that caches block filters from the index store to allow efficient iteration over block filters in ascending order.
+/// </summary>
+public class BlockFilterIterator
+{
+	public BlockFilterIterator(IIndexStore indexStore, int maxNumberFiltersInMemory = 1000)
```

Comment:
Just out of curiosity on the style, can you explain why you prefer this instead of a `const`? thanks.